src/plugins/intel_npu/src/al/include/intel_npu/al/icompiler.hpp

```diff
@@ -112,6 +112,9 @@ struct NetworkMetadata final {
 
     size_t numStreams = 1;
 
+    // Used primarily in the CID path to pass the level zero graph handle from compiler to the backend executor
+    void* graphHandle = nullptr;
```

Comment:
It's not a good idea to extend this structure. No code used in the compiler should be aware of L0 and zero wrappers. We need to pass it separately just for zero compiler adapter.

Reply:
Shall not name graphHandle here, but we need NetworkMetadata to provide room for each compiler to fill private data, ICompiler shall provide this feature

Reply:
We agreed that we will move this outside of the metadata used by CIP. We will have to do it in a different PR since we need extra abstraction layer between the plugin and the compiler.